lib/saluki-components/src/sources/checks/listener.rs

```diff
@@ -0,0 +1,282 @@
+use tokio::fs;
+
+use super::*;
+
+pub struct DirCheckRequestListener {
```

Comment:
Copied from https://github.com/DataDog/saluki/pull/611#discussion_r2042255043

Scoot: I envision this entire DirCheckRequestListener to either go away, or become backed by a grpc subscription to the core-agent's check-scheduler